src/components/Table/index.test.tsx

```diff
@@ -0,0 +1,169 @@
+        extraLargeHiddenColumns={extraLargeHiddenColumns}
+      />
+    )
+
+    expect(screen.getByText(ColumnHeader.Column1)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column2)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column3)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column4)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column5)).toBeInTheDocument()
+  })
+
+  it('shows all columns for large breakpoint', () => {
+    window.innerWidth = 1024
+
+    render(
+      <Table
+        columns={columns}
+        data={data}
+        smallHiddenColumns={smallHiddenColumns}
+        mediumHiddenColumns={mediumHiddenColumns}
+        largeHiddenColumns={largeHiddenColumns}
+        extraLargeHiddenColumns={extraLargeHiddenColumns}
+      />
+    )
+
+    expect(screen.getByText(ColumnHeader.Column1)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column2)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column3)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column4)).toBeInTheDocument()
+  })
+
+  it('hides some columns at medium breakpoint', () => {
+    window.innerWidth = 768
+
+    render(
+      <Table
+        columns={columns}
+        data={data}
+        smallHiddenColumns={smallHiddenColumns}
+        mediumHiddenColumns={mediumHiddenColumns}
+        largeHiddenColumns={largeHiddenColumns}
+        extraLargeHiddenColumns={extraLargeHiddenColumns}
+      />
+    )
+
+    expect(screen.getByText(ColumnHeader.Column1)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column2)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column3)).toBeInTheDocument()
+  })
+
+  it('hides some columns at small breakpoint', () => {
+    window.innerWidth = 640
+
+    render(
+      <Table
+        columns={columns}
+        data={data}
+        smallHiddenColumns={smallHiddenColumns}
+        mediumHiddenColumns={mediumHiddenColumns}
+        largeHiddenColumns={largeHiddenColumns}
+        extraLargeHiddenColumns={extraLargeHiddenColumns}
+      />
+    )
+
+    expect(screen.getByText(ColumnHeader.Column1)).toBeInTheDocument()
+    expect(screen.getByText(ColumnHeader.Column2)).toBeInTheDocument()
+  })
+
+  it('displays row of data', () => {
+    window.innerWidth = 1280
+
+    render(
+      <Table
+        columns={columns}
+        data={data}
+        smallHiddenColumns={smallHiddenColumns}
+        mediumHiddenColumns={mediumHiddenColumns}
+        largeHiddenColumns={largeHiddenColumns}
+        extraLargeHiddenColumns={extraLargeHiddenColumns}
+      />
+    )
+
+    expect(screen.getByText('column1-data')).toBeInTheDocument()
+    expect(screen.getByText('column2-data')).toBeInTheDocument()
+    expect(screen.getByText('column3-data')).toBeInTheDocument()
+    expect(screen.getByText('column4-data')).toBeInTheDocument()
+    expect(screen.getByText('column5-data')).toBeInTheDocument()
+  })
+})
```

Comment:
I think it's best for each unit test to test only one thing. This immediately makes it clear what's wrong when a test starts failing, instead of having to look into the test code to understand which part of the test failed.